### PR TITLE
[lldb] Add null checks to childAtPath helper function

### PR DESCRIPTION
**Pull request by kastiglione**
Add conservative null checks to avoid expected input data.

rdar://77286761

### CODE REVIEW

**Comment by kastiglione** [comment]
@swift-ci test

**Comment by kastiglione** [comment]
@swift-ci test

File: lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeNames.cpp [hunk 1]
```diff
@@ -60,12 +60,12 @@ enum class ThunkAction {
 static swift::Demangle::NodePointer
 childAtPath(swift::Demangle::NodePointer node,
             llvm::ArrayRef<swift::Demangle::Node::Kind> path) {
-  if (path.empty())
+  if (!node || !node->hasChildren() || path.empty())
```
Comment (adrian-prantl) [inline lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeNames.cpp]: nit: `!node->hasChildren()` is redundant with `for (auto *child : *node)`

**Review by adrian-prantl — APPROVED** [review]

File: lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeNames.cpp (hunk 1 above)
Comment (kastiglione) [reply lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeNames.cpp]: thanks

**Comment by kastiglione** [comment]
@swift-ci test

**Pull request merged**